src/wallet-account.ts

```diff
@@ -230,6 +230,10 @@ class ConnectedWalletAccount extends Account {
 
         if (permission.FunctionCall) {
             const { receiver_id: allowedReceiverId, method_names: allowedMethods } = permission.FunctionCall;
+            // check if access_key has 2fa method
+            if (allowedReceiverId === this.accountId && allowedMethods.includes('add_request_and_confirm')) {
```

Comment:
This may be a question to clarify in a tutorial/how to doc: does the contract have to implement add_request_and_confirm?

Reply:
expanded comment